hbase-hadoop2-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationGlobalSourceSource.java

```diff
@@ -197,6 +205,20 @@ public void incrFailedRecoveryQueue() {
     failedRecoveryQueue.incr(1L);
   }
 
+  @Override
+  public void setPeerZkConnectionFailures(boolean success) {
```

Comment:
Since this is a numeric metric, we should comply with the other numeric metrics and provide an _incrPeerZkConnectionFailures_ method, rather than a _setter_.

Reply:
@wchevreuil This was intentional since we want to reset it to zero once we get the success connection. 
Do you suggest having a separate method for a reset?

Reply:
This logic in a global source doesn't give the intended result here because `peerZkConnectionFailures` is reset to 0 if _any_ source succeeds. That doesn't seem right.